### PR TITLE
Methods to Guild and User

### PR DESCRIPTION
**Pull request by Endercheif**
### Changes

-   `adds`: These methods
    - `Client.get_user`
    - `Guild.get_member`
    - `Guild.modify_member`
    - `User.from_id`

> Note: This will not work without #72 being merged

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

**Comment by Endercheif** [comment]
Codacy says "method already defined line 519"

for that method, the duplicate is just a function that is typed with the `@typing.overload` decorator. This is to increase the quality of the code since the function relies on a dictionary and by using `**kwargs` instead of listing it, the arguments it can be directly passed in to the api.

File: pincer/objects/guild.py [hunk 1]
```diff
@@ -499,6 +504,28 @@ async def from_id(cls, client: Client, _id: int) -> Guild:
...
+
+        :param _id:
+            The id of the guild member which should be fetched from the Discord
+            gateway.
+
+        """
+        return await GuildMember.from_id(self._client, self.id, _id)
+
+    @overload
+    async def modify_member(self, *, _id: int, nick: Optional[str] = None, roles: Optional[List[Snowflake]] = None, mute: Optional[bool] = None, deaf: Optional[bool] = None, channel_id: Optional[Snowflake] = None) -> GuildMember:
```
Comment (Sigmanificient) [inline pincer/objects/guild.py:519]: Please ensure pep8 & E501 with 80 characters wrapping

File: pincer/objects/guild.py [hunk 2]
```diff
@@ -248,6 +248,10 @@ class Guild(APIObject):
     """
     Represents a Discord guild/server in which your client resides.
 
+    :param _client:
```
Comment (Sigmanificient) [inline pincer/objects/guild.py:251]: > Can you fill the params for `_client` & `_http`?

File: pincer/objects/guild.py [hunk 3]
```diff
@@ -499,6 +504,28 @@ async def from_id(cls, client: Client, _id: int) -> Guild:
...
         return Guild(**data) 
 
+    async def get_member(self, _id: int):
+        """
+        Fetches a GuildMember from its identifier
+
+        :param _id:
+            The id of the guild member which should be fetched from the Discord
+            gateway.
+
```
Comment (Sigmanificient) [inline pincer/objects/guild.py:514]: > Can you add the return info with the __doc__ please?

File: pincer/objects/user.py [hunk 4]
```diff
@@ -53,6 +57,10 @@ class User(APIObject):
     Represents a Discord user. This can be a bot account or a
     human account.
 
+    :param _client:
```
Comment (Sigmanificient) [inline pincer/objects/user.py:60]: > Can you fill the params for `_client` & `_http`?

File: pincer/objects/user_message.py [hunk 5]
```diff
@@ -174,6 +179,10 @@ class UserMessage(APIObject):
     """
     Represents a message sent in a channel within Discord.
 
+    :param _client:
```
Comment (Sigmanificient) [inline pincer/objects/user_message.py:182]: > Can you fill the params for `_client` & `_http`?

File: pincer/utils/conversion.py [hunk 6]
```diff
@@ -38,7 +42,7 @@ def handle_factory() -> T:
         def fin_fac(v: Any):
             return (
                 v if check is not None and isinstance(v, check)
-                else factory(v)
+                else factory(v) if client is None else factory(v | {"_client": client, "_http": client.http})
```
Comment (Sigmanificient) [inline pincer/utils/conversion.py:45]: > pep8 & 501 also don't apply here

Comment (Sigmanificient) [reply pincer/utils/conversion.py:45]: Tho i would chain ternary as it make the readibility super hard

File: pincer/middleware/error.py [hunk 7]
```diff
@@ -1,43 +1,45 @@
...
+def error_middleware(self, payload: GatewayDispatch):
+    """
+    Middleware for ``on_error`` event.
+
+    :param client:
+
+    :param payload:
+        The data received from the ready event.
+    """
+    return "on_error", [DiscordError.from_dict({"_client": self, "_http": self.http} | payload.data)]
```
Comment (Sigmanificient) [inline pincer/middleware/error.py:41]: pep8 & 501 also don't apply here

may want to put 
```py
return "on_error",  [
    DiscordError.from_dict(
        {"_client": self, "_http": self.http} | payload.data
    )
]

```

File: pincer/middleware/interaction_create.py [hunk 8]
```diff
@@ -48,7 +48,7 @@ async def interaction_create_middleware(self, payload: GatewayDispatch):
     :param payload:
         The data received from the interaction event.
     """
-    interaction: Interaction = Interaction.from_dict(payload.data)
+    interaction: Interaction = Interaction.from_dict(payload.data | {"_client": self, "_http": self.http })
```
Comment (Sigmanificient) [inline pincer/middleware/interaction_create.py:51]: pep8 & 501 also don't apply here

may want to put
```py
interaction: Interaction = Interaction.from_dict(
    payload.data | {"_client": self, "_http": self.http }
)
```

File: pincer/middleware/message_delete.py [hunk 9]
```diff
@@ -1,45 +1,45 @@
...
+    """
+    Middleware for ``on_message_delete`` event.
+
+    :param self:
+        The current client.
+
+    :param payload:
+        The data received from the delete message event.
+    """
+    return "on_message_delete", [MessageDeleteEvent.from_dict({"_client": self, "_http": self.http} | payload.data)]
```
Comment (Sigmanificient) [inline pincer/middleware/message_delete.py:41]: pep8 & 501 also don't apply here

may want to put 
```py
return "on_message_delete", [
    MessageDeleteEvent.from_dict(
        {"_client": self, "_http": self.http} | payload.data
    )
]

```

File: pincer/middleware/ready.py [hunk 10]
```diff
@@ -1,51 +1,51 @@
...
+    """
+    Middleware for ``on_ready`` event.
+
+    :param self:
+        The current client.
+
+    :param payload:
+        The data received from the ready event.
+    """
+    self.bot = User.from_dict({"_client": self, "_http": self.http} | payload.data.get("user"))
```
Comment (Sigmanificient) [inline pincer/middleware/ready.py:45]: pep8 & 501 also don't apply here

may want to put 
```py
self.bot = User.from_dict(
    {"_client": self, "_http": self.http} | payload.data.get("user")
)
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pincer/objects/guild.py [hunk 11]
```diff
@@ -453,6 +457,7 @@ class Guild(APIObject):
     approximate_member_count: APINullable[int] = MISSING
     approximate_presence_count: APINullable[int] = MISSING
     channels: APINullable[List[Channel]] = field(default_factory=list)
+    hub_type: APINullable[...] = MISSING # TODO: Add type when type is known
```
[the remark this answers is not in the document]
Comment (Endercheif) [reply pincer/objects/guild.py:460]: It is not in the docs but comes in the api. I assume it's new

[the remark this answers is not in the document]
Comment (Endercheif) [reply pincer/objects/guild.py:460]: similarly, `nsfw` is not in the docs but comes in the api

**Review by Sigmanificient — APPROVED** [review]

**Pull request merged**